R/rdeephaven/R/helper_functions.R

```diff
@@ -3,7 +3,11 @@ first_class <- function(arg) {
 }
 
 verify_type <- function(arg_name, candidate, required_type, message_type_name, is_scalar) {
-  if (!is_scalar && (first_class(candidate) == "list")) {
+  if (required_type == "list") {
+    if (first_class(candidate) != "list") {
+      stop(paste0("'", arg_name, "' must be a list or a vector of lists. Got an object of class ", first_class(candidate), "."))
```

Comment:
```suggestion
      stop(paste0("'", arg_name, "' must be a list. Got an object of class ", first_class(candidate), "."))
```
Is the removed text always appropriate?

Reply:
I believe so. As far as R is concerned, there is no difference between `list(a = "val", b = "val")` and `c(list(a = "val"), list(b = "val"))`. so we cannot programmatically distinguish between them. So, the "or a vector of lists" is technically what we will accept for any list-valued argument.

Reply:
```suggestion
      stop(paste0("'", arg_name, "' must be a list or a vector. Got an object of class ", first_class(candidate), "."))
```